source/threads.tex

```diff
@@ -2838,18 +2838,20 @@
 all objects with thread storage duration associated with the current thread.
 
 \pnum
-\realnote The supplied lock will be held until the thread exits, and care
+\begin{note}
+The supplied lock will be held until the thread exits, and care
 must be taken to ensure that this does not cause deadlock due to lock
 ordering issues. After calling \tcode{notify_all_at_thread_exit} it is
 recommended that the thread should be exited as soon as possible, and
 that no blocking or time-consuming tasks are run on that thread.
 
 \pnum
-\realnote It is the user's responsibility to ensure that waiting threads
```

Comment:
Use two separate notes here? Generally I'd like to avoid `\pnum`s within a note.